domain/src/main/java/org/oppia/android/domain/spotlight/SpotlightStateController.kt

```diff
@@ -79,14 +83,14 @@ class SpotlightStateController @Inject constructor(
         RETRIEVE_SPOTLIGHT_CHECKPOINT_DATA_PROVIDER_ID
       ) {
         val viewState = when (feature) {
-          ONBOARDING_NEXT_BUTTON -> it.onboardingNextButton
-          TOPIC_LESSON_TAB -> it.topicLessonTab
-          TOPIC_REVISION_TAB -> it.topicRevisionTab
-          FIRST_CHAPTER -> it.firstChapter
-          PROMOTED_STORIES -> it.promotedStories
-          LESSONS_BACK_BUTTON -> it.lessonsBackButton
-          VOICEOVER_PLAY_ICON -> it.voiceoverPlayIcon
-          VOICEOVER_LANGUAGE_ICON -> it.voiceoverLanguageIcon
+          ONBOARDING_NEXT_BUTTON -> spotlightStateRetrieverHelper(it.onboardingNextButton)
```

Comment:
I suggest making the change down when viewState is being returned so that it only needs to happen in one place.

Reply:
done